### PR TITLE
Handle flatpak+https scheme

### PR DESCRIPTION
**Pull request by meisenzahl**
Closes https://github.com/elementary/sideload/issues/169

### CODE REVIEW

**Comment by meisenzahl** [comment]
I use [GLib.File.copy](https://valadoc.org/gio-2.0/GLib.File.copy.html) to download the FlatpakRef. This was the easiest way without adding new external dependencies.

File: src/Application.vala [hunk 1]
```diff
@@ -54,6 +56,30 @@ public class Sideload.Application : Gtk.Application {
...
+            var remote_file = File.new_for_uri (uri);
+            File local_file = File.new_for_path (path);
+            try {
+                if (!remote_file.copy (local_file, FileCopyFlags.OVERWRITE)) {
+                    warning ("Failed to download file from %s", uri);
+                    release ();
+                    return;
+                }
+            } catch (Error e) {
+                critical ("Failed to download file from %s: %s", uri, e.message);
```
Comment (danirabbit) [inline src/Application.vala:75]: I think we probably need to throw an error that the user can see in case of a failure here

**Review by danirabbit — CHANGES_REQUESTED** [review]
Works as expected, just one request to make sure we throw an error dialog or I can't remember if Sideload has an error view. Something that the user can see the error

**Comment by meisenzahl** [comment]
In order to be able to display an error to a user in the `ErrorView`, I unfortunately had to move the logic for parsing the Flatpak file to the `MainWindow`. @danirabbit let me know what you think.

**Review by danirabbit — APPROVED** [review]
Hey nice. I didn't realize that so many of these other errors were silent as well. Thanks for fixing those! Seems like everything is working as expected. Good job!

**Pull request merged**